crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -714,34 +721,43 @@ impl ArtifactFetcher {
     /// (because multiple files can share one [`DebugId`]).
     #[tracing::instrument(skip(self))]
     pub async fn get_file(&mut self, key: &FileKey) -> CachedFileEntry {
-        if let Some(file) = self.try_get_file_using_index(key).await {
-            return file;
+        let mut file = self.try_get_file_using_index(key).await;
+
+        if file.is_none() {
+            // Try looking up the file in one of the artifact bundles that we know about.
+            file = self.try_get_file_from_bundles(key);
         }
 
-        // Try looking up the file in one of the artifact bundles that we know about.
-        if let Some(file) = self.try_get_file_from_bundles(key) {
-            return file;
+        if file.is_none() {
+            // Otherwise, try to get the file from an individual artifact.
+            file = self.try_fetch_file_from_artifacts(key).await;
         }
 
-        // Otherwise, try to get the file from an individual artifact.
-        if let Some(file) = self.try_fetch_file_from_artifacts(key).await {
-            return file;
+        if file.is_none() {
+            // Otherwise: Do a (cached) API lookup for the `abs_path` + `DebugId`
+            if self.query_sentry_for_file(key).await {
+                // At this point, *one* of our known artifacts includes the file we are looking for.
+                // So we do the whole dance yet again.
+                file = self.try_get_file_from_bundles(key);
+                if file.is_none() {
+                    file = self.try_fetch_file_from_artifacts(key).await;
+                }
+            }
         }
 
-        // Otherwise: Do a (cached) API lookup for the `abs_path` + `DebugId`
-        if self.query_sentry_for_file(key).await {
-            // At this point, *one* of our known artifacts includes the file we are looking for.
-            // So we do the whole dance yet again.
-            if let Some(file) = self.try_get_file_from_bundles(key) {
-                return file;
+        match file {
+            Some(file) => {
```

Comment:
I would rather use `if let` and early-return.

In general I hope the compiler is smart enough to optimize all of these checks away in the sense that it jumps over all the `is_none` checks when the file is actually found early.

Reply:
You mean just for this block or in general in the function?

Reply:
`if let` for this block, yes.

in general this looks good, as I don’t know how we could retain the early-return behavior otherwise.

Reply:
Yeah I'm honestly not too thrilled about changing this function, but I can't figure out a better way to do it. I'll change this block though.